core/src/main/java/fr/sncf/osrd/utils/graph/Pathfinding.java

```diff
@@ -68,6 +68,8 @@ public record Range(double start, double end){}
     private AStarHeuristic<EdgeT> estimateRemainingDistance = null;
     /** Function to call to know the cost of the range. */
     private EdgeRangeCost<EdgeT> edgeRangeCost = null;
+    /** Function to call to know the cost of the edge location. */
+    private TotalDistanceUntilEdgeLocation<EdgeT> totalDistanceUntilEdgeLocation = null;
```

Comment:
We could add an assert in `checkParameters`, we need exactly one of `totalDistanceUntilEdgeLocation` and `edgeRangeToCost` to be set. 
We could also add more documentation about this.